massgov/openmass
# DP-14197 Release trigger 2.0

**Description:**
This was work in progress back in late November I just never was able to finish it until now. I think everything is working now. This will help with not having to turn the release automation off during the holidays and it won't run unless a PR with a changelog file has been merged into develop. 
- [x] Checks for holidays and compares to the today date to see if a release branch should be cut. (New Years, July 4th, Veterans Day, and Christmas Day) It will not check for Thanksgiving because no definite date. All other holidays are on Mondays used https://www.sec.state.ma.us/cis/cispdf/ma_legal_holiday.pdf)
- [x] Checks to see if any changelogs in the changelog directory.
- [x] Updates the body of a PR for the release branch and the body of the GitHub tag

**Jira:**
https://jira.mass.gov/browse/DP-14197

**To Test:**
- [x] You will only be able to test the changelogs and holiday piece at the moment. (I left a few lines in there to check the holiday part)
- [x] Comment out everything under `// Creates the release branch with new version.'
- [x] Within the terminal run the following `php scripts/build-changelog.php` -> in the terminal you will see today date update to a holiday here `// Test the holiday section this can be removed after testing done.` 
- [x] Run the following command again `php scripts/build-changelog.php` it should give you `No release today because either nothing to release or today is a holiday.`
- [x] Change to a non-holiday date and run the command again `php scripts/build-changelog.php`. It should say `The release branch will continue.` and you should see the following `Display new version for release:0.227.0`
- [x] Delete all changelogs next and run the following command `php scripts/build-changelog.php` it should say the following `No release today because either nothing to release or today is a holiday.`

**NOTE:** Once testing all done uncomment everything below the `// Creates the release branch with new version.' and remove test date lines 23-29; Also uncomment lines 17-21 this is telling the script what today date is. 

@weitzman and @yriahi could one of you check the bash script for tagging the GitHub I've update to use ` "body": cat /scripts/changelog-body.txt >` I'm not sure if this is going to work correctly for cURL.

If you guys want me to clean this up after testing has been done let me know.

**Screenshots/GIFs:**
Use something like [licecap](http://www.cockos.com/licecap/) to capture gifs to demonstrate behaviors.

---

[Peer Review Checklist](https://github.com/massgov/opemass/blob/develop/docs/peer_review_checklist.md)
